warehouse/integrations/azure-synapse/azure-synapse.go

```diff
@@ -455,15 +456,15 @@ func (as *AzureSynapse) loadUserTables() (errorMap map[string]error) {
 		if colName == "id" {
 			continue
 		}
-		userColNames = append(userColNames, colName)
+		userColNames = append(userColNames, fmt.Sprintf(`%q`, colName))
 		caseSubQuery := fmt.Sprintf(`case
 						  when (exists(select 1)) then (
-						  	select top 1 %[1]s from %[2]s
+						  	select top 1 %[1]q from %[2]s
 						  	where x.id = %[2]s.id
-							  and %[1]s is not null
+							  and %[1]q is not null
 							order by X.received_at desc
 							)
-						  end as %[1]s`, colName, as.Namespace+"."+unionStagingTableName)
+						  end as %[1]q`, colName, as.Namespace+"."+unionStagingTableName)
```

Comment:
This got missed when we added support for skipping escaping keywords. https://github.com/rudderlabs/rudder-server/pull/2115